### PR TITLE
don't parse response string if we have a success response

### PR DESCRIPTION
**Pull request by nwadams**
don't need to parse errorResponseString on a successful response

### CODE REVIEW

File: src/com/android/volley/NetworkResponse.java [hunk 1]
```diff
@@ -39,14 +39,19 @@ public NetworkResponse(int statusCode, byte[] data, Map<String, String> headers,
...
+        if (statusCode != HttpStatus.SC_OK) {
+            String inter;
+            try {
+                inter = new String(data, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                inter = "<error parse failed>";
+            }
+            errorResponseString = inter;
+        } else {
+            errorResponseString = null;
```
Comment (carolineleung) [inline src/com/android/volley/NetworkResponse.java:53]: Can't we just set errorResponseString directly instead of using the extra inter string?

Comment (nwadams) [inline src/com/android/volley/NetworkResponse.java:53]: errorResponseString is final

**Comment by carolineleung** [comment]
:+1:

**Comment by lxdvs** [comment]
ship it

**Pull request merged**